### PR TITLE
added `ifftn` to ivy functional api

### LINKED ISSUE #17393: ifftn

### PR DESCRIPTION
**Pull request by akshatvishu**
Close #17393

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Great job @akshatvishu 🚀

I've left some formatting comments, otherwise everything should be good to go after we run the test

**Comment by akshatvishu** [comment]
Please let it 

Dog LGTM Keyboard

**Review by KareemMAX — CHANGES_REQUESTED** [review]
> Please let it
> 
> Dog LGTM Keyboard [ Dog LGTM Keyboard ](https://camo.githubusercontent.com/21496c7fd6e2f0584a5ebcc11f16becfaa307751a2483c29079a050c098b3eaf/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6b62567a454e564e6c4536304d51523148592f67697068792e676966) [ ](https://camo.githubusercontent.com/21496c7fd6e2f0584a5ebcc11f16becfaa307751a2483c29079a050c098b3eaf/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6b62567a454e564e6c4536304d51523148592f67697068792e676966)

Wish I could 😅. As we are trying to adhere to [PEP 257](https://peps.python.org/pep-0257/#multi-line-docstrings) I need the issues you resolved to be fixed before merging. In the meantime I'm reviewing the `ifftn` input space to make sure that your test is exhaustive, if it is and it passes the PR will be good to go.

**Comment by KareemMAX** [comment]
Just a quick side note, all tests have passed. Once you make the docstring changes your PR is good to merge

**Comment by akshatvishu** [comment]
> Wish I could sweat_smile. As we are trying to adhere to [PEP 257](https://peps.python.org/pep-0257/#multi-line-docstrings) I need the issues you resolved to be fixed before merging. In the meantime I'm reviewing the ifftn input space to make sure that your test is exhaustive, if it is and it passes the PR will be good to go.

Ahh, So sorry @KareemMAX  I found where the mistake is happening. When I try to change the `docstring` as you suggested  the  `pre-commit` hook for `docformatter` fails.

image

And then this hook reformat my doc to its current state back again from the changes that you suggested and thus, changes you suggested aren't reflected when I push the code.
I was so confused because  I did make the changes that you suggested!  Should I skip the `pre-commit` hook entirely to apply changes that you suggested?

**Comment by akshatvishu** [comment]
I made the changes to doc  as you suggested by skipping `pre-commit`

**Comment by KareemMAX** [comment]
@akshatvishu I'm guessing you have missed the blank line between summary and description. Which confused pre-commit

File: ivy/data_classes/container/experimental/layers.py
Comment (KareemMAX) [inline ivy/data_classes/container/experimental/layers.py:1925]: Seems like I made a wrong code suggestion 😅
```suggestion
        ivy.Container static method variant of ivy.ifftn. 
        
```

File: ivy/data_classes/container/experimental/layers.py
Comment (KareemMAX) [inline ivy/data_classes/container/experimental/layers.py:1983]: Same error here
```suggestion
        ivy.Container static method variant of ivy.ifftn. 
        
```

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Seems like it was an issue from my side 😅😅

Just apply those suggestions and commit them view pre-commit and you are good to go

Sorry 😅😅

**Review by KareemMAX — APPROVED** [review]
LGTM! finally 😂

Merging now

**Comment by akshatvishu** [comment]
> LGTM! finally joy
> 
> Merging now

Kid Pumped

**Pull request merged**
